shared-libs/cht-datasource/src/local/target-interval.ts

```diff
@@ -38,4 +39,100 @@ export namespace v1 {
       return doc;
     };
   };
+
+  /** @internal */
+  export const getPage = ({ medicDb: db }: LocalDataContext) => {
+    return async (
+      qualifier: (ReportingPeriodQualifier & ContactUuidsQualifier),
+      cursor: Nullable<string>,
+      limit: number,
+    ): Promise<Page<TargetInterval.v1.TargetInterval>> => {
+      
+      const skip = validateCursor(cursor);
+      const intervalIds = await fetchTargetIntervalIds(limit, skip)(db, qualifier);
+
+      return await fetchTargetIntervalDocs(limit, skip)(db, intervalIds.data);
+    };
+  };
+  
+  const fetchTargetIntervalIds = (limit: number, skip: number) => async (
+    db: PouchDB.Database, 
+    qualifier: (ReportingPeriodQualifier & ContactUuidsQualifier)
+  ) => {
+    const fetchFn = async (
+      limit: number,
+      skip: number
+    ) => {
+      const result = await db
+        .allDocs({ include_docs: false, limit, skip, startkey: `target~${qualifier.reportingPeriod}` });
+      
+      if (result.rows.length === 0) {
+        return [];
+      }
+
+      return result.rows
+        .filter(row => row.doc)
+        .map(row => row.doc as TargetInterval.v1.TargetInterval);
```

Comment:
```suggestion
      const result = await db.allDocs({
        include_docs: false,
        limit,
        skip,
        startkey: `target~${qualifier.reportingPeriod}~`,
        endkey: `target~${qualifier.reportingPeriod}~\ufff0`,
      });
      return result.rows.map(({ id }) => id);
```

This was super close here, but there are a couple problems. First, with `include_docs: false` we are not going to get anything back for `row.doc`. This is fine because we actually only want `row.id`.  This whole `fetchAndFilter` operation can operate on just the UUIDs.

We also need to include the `endkey` option here or Couch will just return _every_ doc that comes alphabetically after our startkey.  The `\ufff0` char is basically the "max value" indicator so we select all the ids with the given reportingPeriod and no others.